MIGRATING.md

```diff
@@ -0,0 +1,134 @@
+
+#### `handleAuthentication` is replaced by `handleLoginRedirect`
+
+`handleLoginRedirect` is called by the `LoginCallback` component as the last step of the login redirect authorization flow. It will obtain and store tokens and then call `restoreOriginalUri` which will return the browser to the `originalUri` which was set before the login redirect flow began.
+
+#### `setFromUri` and `getFromUri` have been replaced with `setOriginalUri` and `getOriginalUri`
+
+[setOriginalUri](https://github.com/okta/okta-auth-js#setoriginaluriuri) is used to save the current/pending URL before beginning a redirect flow. There is a new option, [restoreOriginalUri](https://github.com/okta/okta-auth-js#restoreoriginaluri), which can be used to customize the last step of the login redirect flow.
+
+#### `isAuthenticated` will be true if **both** accessToken **and** idToken are valid
+
+If you have a custom `isAuthenticated` function which implements the default logic, you may remove it.
+
+#### `getAccessToken` and `getIdToken` have been changed to synchronous methods
+
+With maintaining in-memory [AuthState][] since [@okta/okta-auth-js][] version 4.1, token values can be accessed in synchronous manner.
+
+#### `getTokenManager` has been removed
+
+You may access the `TokenManager` with the `tokenManager` property:
+
+```javascript
+const tokens = $auth.tokenManager.getTokens();
+```
+
+#### `authRedirectGuard` has been removed
+
+Guard logic is handled internally in `@okta/okta-vue@3.x`, previous global guard registration should be removed:
+
+```diff
+- router.beforeEach(Vue.prototype.$auth.authRedirectGuard())
+```
+
+### "Active" token renew
+
+Previously, tokens would only be renewed when they were read from storge. This typically occurred when a user was navigating to a protected route. Now, tokens will be renewed in the background before they expire. If token renew fails, the [AuthState][] will be updated and `isAuthenticated` will be recalculated. If the user is currently on a protected route, they will need to re-authenticate. Set the `onAuthRequired` option to customize behavior when authentication is required. You can set [tokenManager.autoRenew](https://github.com/okta/okta-auth-js/blob/master/README.md#autorenew) to `false` to disable active token renew logic.
+
+### `isAuthenticated` is called by `transformAuthState`
+
+After the [AuthState][] is updated, but before it is emitted, [transformAuthState][] will be called. During this call, the `isAuthenticated` option, if set on the config object, will be called to set the value of `authState.isAuthenticated`. By default, `authState.isAuthenticated` will be true if **both** the access token and ID token are valid. This logic can be customized by providing a custom `isAuthenticated` function on the config object. You may also provide your own [transformAuthState][] function to customize the [AuthState][] object before it is emitted.
```

Comment:
I know this is confusing either way, but do you think `authState` is better to mention instead of `AuthState`?

Reply:
I would use `AuthState` when referring to the type, and `authState` if referring to a function parameter or variable with that name. In this case there is no variable or parameter named `authState`. So maybe instead of "set the value of authState.isAuthenticated", we can do something like "set the value of isAuthenticated on the current AuthState object"